lxd/container.go

```diff
@@ -397,7 +399,7 @@ func containerValidDevices(db *db.Node, devices types.Devices, profile bool, exp
 				return fmt.Errorf("Unix device entry is missing the required \"source\" or \"path\" property.")
 			}
 
-			if m["major"] == "" || m["minor"] == "" {
+			if shared.IsTrue(m["required"]) && (m["major"] == "" || m["minor"] == "") {
```

Comment:
Shouldn't that also consider the case where m["required"] is an empty string (in which case it should be treated as true)?

Reply:
added